rollup/transaction_ingestion.go

```diff
@@ -0,0 +1,83 @@
+package rollup
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/jmoiron/sqlx"
+)
+
+type TxIngestion struct {
+	loopTicker *time.Ticker
+	db         *sqlx.DB
+	signer     types.Signer
+	txpool     *core.TxPool
+}
+
+// Should this have a safety check on cfg.TxIngestionPollInterval?
+func NewTxIngestion(cfg Config, chaincfg *params.ChainConfig, txpool *core.TxPool) *TxIngestion {
+	txIngestion := TxIngestion{
+		signer:     types.NewOVMSigner(chaincfg.ChainID),
+		txpool:     txpool,
+		loopTicker: time.NewTicker(cfg.TxIngestionPollInterval),
+	}
+
+	if cfg.IsTxIngestionEnabled() {
+		conn := txIngestion.makeConn(&cfg)
+		db, err := sqlx.Connect("postgres", conn)
+		if err != nil {
+			log.Error("Cannot connect to postgres", "msg", err.Error())
+			return nil
+		}
+
+		txIngestion.db = db
+
+		go txIngestion.loop()
+	}
+
+	return &txIngestion
+}
+
+func (t *TxIngestion) applyTransaction(tx *types.Transaction) error {
+	return t.txpool.AddLocal(tx)
+}
+
+// It would be nice to block on a channel that will
+// unblock when the transaction finishes computing the
+// state transition.
+func (t *TxIngestion) loop() {
+	for range t.loopTicker.C {
+		tx, err := GetMostRecentQueuedTransaction(t.db)
```

Comment:
It would feel safer if there was a check here on the size of the mempool, so that the mempool never fills up and removes transactions. Transactions are inserted as locals, need to double check about local eviction. I believe that external txs (from p2p network) are evicted before local txs and that local txs can still be evicted when the mempool is completely full.